code/lib/preview-api/src/modules/preview-web/PreviewWeb.test.ts

```diff
@@ -2735,6 +2735,60 @@ describe('PreviewWeb', () => {
       });
     });
 
+    describe('if called twice simultaneously', () => {
+      it('does not get renders confused', async () => {
+        const [importGate, openImportGate] = createGate();
+        const [importedGate, openImportedGate] = createGate();
```

Comment:
nit: could you name these more clearly? It took me a few passes before I understood what was going on.